### ISSUE TITLE
No automatic early stopping

### ISSUE THREAD

**Issue by AndyMcAliley** [USGS-R/lake-temperature-lstm-static#35]
During training, we want to save the set of model weights that corresponds to the training epoch with the lowest validation loss. Also, we want to stop training once we're confident that the validation loss isn't going to decrease more. Loss values tend to bounce around from epoch to epoch, so we'll want to give it a few epochs rather than stop after the first epoch when validation loss increases. So, to implement early stopping,

1. Save model weights for the lowest validation loss so far.
2. Stop training if validation loss hasn't decreased for N epochs.

**Comment by AndyMcAliley** [comment]
Closed by #40
